vcd/resource_vcd_edgegateway.go

```diff
@@ -0,0 +1,201 @@
+package vcd
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func resourceVcdEdgeGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdEdgeGatewayCreate,
+		Delete: resourceVcdEdgeGatewayDelete,
+		Read:   resourceVcdEdgeGatewayRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"org": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"vdc": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			/*
+					// RFC:
+				    // Waiting on a decision whether such parameters are needed
+					"delete_force": &schema.Schema{
+						Type:        schema.TypeBool,
+						Required:    true,
+						ForceNew:    true,
+						Description: "TBD: no description in the API, and undefined behavior in practice",
+					},
+					"delete_recursive": &schema.Schema{
+						Type:        schema.TypeBool,
+						Required:    true,
+						ForceNew:    true,
+						Description: "TBD: no description in the API, and undefined behavior in practice",
+					},
+			*/
+			"advanced_networking": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				ForceNew:    true,
+				Description: "True if the gateway uses advanced networking",
+			},
+			"backing_config": &schema.Schema{
```

Comment:
'backing_config" -> "gateway_configuration" to be line with UI ?

Reply:
No objections. 
@lvirbalas , @Didainius ?

Reply:
+1 for `gateway_configuration`

Reply:
Changed to `gateway_configuration`